lib/datastore.js

```diff
@@ -465,6 +465,12 @@ class Datastore extends EventEmitter {
       err.missingFieldName = true
       throw err
     }
+    if (Array.isArray(options.fieldName)) {
+      options.fieldName.sort()
+    }
+    if ([].concat(options.fieldName).some(field => field.includes(','))) {
+      throw new Error('Cannot use comma in index fieldName')
+    }
     if (this.indexes[options.fieldName]) return
```

Comment:
Even though you explicitly said it in your comment, I'd rather explicitly call `.join(',')`

Reply:
Thank you very much, of course you can push on my branch if you want, do you need access ?

Reply:
Upon further review, I'm not comfortable at all having a fieldName which is sometimes a string and sometimes an Array, it leads to confusion as the fieldName is written on disk when creating or removing an index in persistence.js.

Even though it is slightly sub-optimal I'd rather join fieldName in ensureIndex & removeIndex, and split them in the constructor of Index in a secondary field